### ISSUE TITLE
Port TextMate's “Wrap Selection In Tag”

### ISSUE THREAD

**Issue by steveluscher** [atom/language-html#9]
I regularly mash `ctrl-shift-w` in TextMate; it wraps the current selection in `<p>…</p>`, highlights both ‘p’ characters letting me type a replacement, then unlatches the selection as soon as I hit space (as I continue to type an attribute/value pair). It's pretty awesome.

Anyone up for porting it over?

[TextMate's “Wrap Selection In Tag”](https://github.com/textmate/html.tmbundle/blob/master/Snippets/Wrap%20Selection%20In%20Tag.plist)

**Comment by jjulian** [comment]
@dannymichel Try `ctrl + w` (https://atom.io/packages/emmet)

**Comment by steve-ross** [comment]
emmet isn't at all what I'm looking for, seems way overkill for such a simple feature

**Comment by Trudko** [comment]
@steve-ross  and others I created simple package which might help (Wrap Selection)[https://atom.io/packages/wrap-selection]. Please if something is missing let me know!

**Comment by fricks** [comment]
i have no success :-1:

**Comment by cjmosure** [comment]
Yup, I find myself popping open Sublime instead of Atom for this. The intended functionality should be a shortcut that automatically wraps the selected text in `<p>` and `</p>` with both `p`'s selected for manipulation. Having emmet open a dialog box to choose the text to wrap it in works, but not as quickly as Sublime's approach.

**Comment by jonathanbello** [comment]
Install Emmet and use `ctrl+w` and a small bottom window will pop up then just type your tag and that will work.

**Comment by peterthomson** [comment]
Can't believe Atom lacks such a simple but important feature. And using Emmet to create a clunky multi-step solution is not a solution to this keyboard shortcut issue for wrapping the selection in a tag.

**Comment by stevesunderland** [comment]
Wow can't believe this is still an issue! Back to Sublime :sob:

**Comment by lpdumas** [comment]
As @jonathanbello said, using emmet is pretty straight foward.
![atom](https://cloud.githubusercontent.com/assets/867843/8430722/8cf92bf6-1f01-11e5-9061-94fbffb42b88.gif)

**Comment by cjmosure** [comment]
Sublime skips the input step, which I feel is 1000 times better

![6b3rkxlsll](https://cloud.githubusercontent.com/assets/1966405/8431613/ab2bc212-1f08-11e5-890f-2cd96a075559.gif)

**Comment by da-n** [comment]
Been trying out Atom for a few days and pretty amazed this feature is not included, in Sublime I use it dozens of times a day. Would be awesome not have to install Emmet. Aside from this, Atom is very impressive.

**Comment by DaveEveritt** [comment]
Writing a lot of HTML means, for me, this is essential in one of the Atom core packages `language-html` or `autocomplete-html`.

**Comment by geekish** [comment]
+1. I hit `ctrl-shift-w` way too often... Emmet just isn't my thing.

**Comment by sanusart** [comment]
I was looking for this functionality too as I recently trying Atom (and really liking it) and I just miss simple "mark and wrap" shortcut. I made very simple package to wrap selection in html tag. Don't expect too much from it though [atom-wrap-in-tag](https://atom.io/packages/atom-wrap-in-tag). Not sure I'll have much time to improve it or bugfix it, but maybe someone will find it useful.

**Comment by DaveEveritt** [comment]
@sanusart I like it. Nice and simple, no clashes with other keymappings, does one thing, and does it well. And far less clunky than the Emmet way. Installed :-)

**Comment by jonobr1** [comment]
+1!

**Comment by fricks** [comment]
guys, how select in atom windows version the block of start and end of div or whatever? in mac just cmd + , but in windows i dont figured out how works :/

**Comment by geekish** [comment]
> I was looking for this functionality too as I recently trying Atom (and really liking it) and I just miss simple "mark and wrap" shortcut. I made very simple package to wrap selection in html tag. Don't expect too much from it though atom-wrap-in-tag. Not sure I'll have much time to improve it or bugfix it, but maybe someone will find it useful.

Finally installed this, working just as it should so far! Thank you!

**Comment by tehapo** [comment]
I also implemented my own version of this functionality as the [atom-tag-wrapper](https://atom.io/packages/atom-tag-wrapper) package. It supports multiple selections and also indents multiline selections correctly inside the added tag.

**Comment by cjmosure** [comment]
Both [atom-wrap-in-tag](https://atom.io/packages/atom-wrap-in-tag) and [atom-tag-wrapper](https://atom.io/packages/atom-tag-wrapper) work prefectly.
